tests/cases/compiler/arrowFunctionParsingGenericInObject.ts

```diff
@@ -0,0 +1,23 @@
+// Apparent parser error when first property of returned object is generic function with default type
+const fails1 = () => ({
+    test: <T = undefined>(value: T): T => value,
+    extraValue: () => {},
+})
+
+// Without a default type, it works fine
+const works1 = () => ({
+    test: <T>(value: T): T => value,
+    extraValue: () => {},
+})
+
+// As second property, it works fine
+const works2 = () => ({
+    extraValue: () => {},
+    test: <T = undefined>(value: T): T => value,
+})
+
+// The first property _must_ be a function, though
+const fails2 = () => ({
+    extraValue: '',
+    test: <T = undefined>(value: T): T => value,
+})

```

Comment:
if all these examples succeed where some failed before, but `async` or `yield` would make them fail again, it would be nice to have test for those too.

Reply:
My expectation is that async actually "fixes" things because it gives a strong clue to the parser as to what is and isn't a function. I'll try for some cases; this was a copy paste from the issue.